components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java

```diff
@@ -1396,7 +1396,8 @@ public static void addApplication(ApplicationBean appDefinition, ConfigurationCo
                     usedCartridgeGroups.toArray(new String[usedCartridgeGroups.size()]));
 
         } catch (AutoscalerServiceApplicationDefinitionExceptionException e) {
-            throw new RestAPIException(e);
+            String message = e.getFaultMessage().getApplicationDefinitionException().getMessage();
```

Comment:
Thanks for pointing Malaka! Ideally this should not raise any NPEs because autoscaler service always returns an application definition exception and an error message. However it might be better to add null checks here. @pubudu538 please note!